pkg/providers/docker/docker.go

```diff
@@ -52,6 +55,12 @@ type provider struct {
 	templateBuilder       *DockerTemplateBuilder
 }
 
+// ProviderKubeconfigWriter reads the kubeconfig secret on a docker cluster and copies the contents to a writer.
+type ProviderKubeconfigWriter struct {
+	docker ProviderClient
+	kubeconfig.Writer
```

Comment:
if the 	kubeconfig.Writer wasn't broader than it needs you wouldn't need to embed this here and you could keep it as a unexported member